### PR TITLE
[Test]MetaAPI: add table opration on absent DB

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: ClSlaid <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add tests for MetaAPI in `common/meta/api/src/meta_api_test_suite.rs`.

## Changelog

- Added tests

## Related Issues

issue #4098 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by ClSlaid** [comment]
Hope the CI could run tests for me...
My little laptop cannot hold such a heavy burden with its 14GB RAM.

**Review by ariesdevil — APPROVED** [review]

**Comment by BohuTANG** [comment]
failures:
[784](https://github.com/datafuselabs/databend/runs/5216628993?check_suite_focus=true#step:3:784)

[785](https://github.com/datafuselabs/databend/runs/5216628993?check_suite_focus=true#step:3:785)
---- meta_api_impl::test_meta_embedded_table_create_get_drop stdout ----
[786](https://github.com/datafuselabs/databend/runs/5216628993?check_suite_focus=true#step:3:786)
thread 'meta_api_impl::test_meta_embedded_table_create_get_drop' panicked at 'assertion failed: `(left == right)`
[787](https://github.com/datafuselabs/databend/runs/5216628993?check_suite_focus=true#step:3:787)
  left: `1003`,
[788](https://github.com/datafuselabs/databend/runs/5216628993?check_suite_focus=true#step:3:788)
 right: `1020`',

File: common/meta/api/src/meta_api_test_suite.rs [hunk 1]
```diff
@@ -427,10 +428,57 @@ impl MetaApiTestSuite {
...
+
+            let got = mt.upsert_table_option(plan).await;
+            tracing::debug!("upsert table on unknow db got: {:?}", got);
+
+            assert!(got.is_err());
+            let err = got.unwrap_err();
+            let err = ErrorCode::from(err);
+
+            assert_eq!(unknown_database_code, err.code());
+        }
```
Comment (ClSlaid) [inline common/meta/api/src/meta_api_test_suite.rs:461]: I brainlessly wrote similar tests for every operation (`create`, `get`, `upsert` and `drop`). However, the `upsert` operation is seen to operate on tables by `table_id`, regardless of DBs.
So removing this test should solve the problem.

**Comment by ClSlaid** [comment]
Strange, I can't reproduce locally on my laptop...

图片

**Review by drmingdrmer — APPROVED** [review]
Nice shot!

**Comment by ClSlaid** [comment]
> Nice shot!

the problem is more complex than I think... It came out to be `MetaServiceError` instead.
I'll take a look tomorrow(this morning).

**Comment by drmingdrmer** [comment]
> > Nice shot!
> 
> the problem is more complex than I think... After deletion, a `MetaServiceError` will be thrown from the drop-on-unknown DB test instead. The implementation of `create` and `drop` has a lot in common, but why `create` returns an `UnknowDataBase` while `drop` returns a `MetaServiceError`? This problem has only shown up in tests of `metasrv`, a further looking into is needed... I'll take a look tomorrow (this morning).

That should be incorrect error type that is used when a database is not found. Let me have a look.

**Comment by ariesdevil** [comment]
> 



> > > Nice shot!
> > 
> > 
> > the problem is more complex than I think... After deletion, a `MetaServiceError` will be thrown from the drop-on-unknown DB test instead. The implementation of `create` and `drop` has a lot in common, but why `create` returns an `UnknowDataBase` while `drop` returns a `MetaServiceError`? This problem has only shown up in tests of `metasrv`, a further looking into is needed... I'll take a look tomorrow (this morning).
> 
> That should be incorrect error type that is used when a database is not found. Let me have a look.

Fixed at: [`d7d8f0b` (#4181)](https://github.com/datafuselabs/databend/pull/4181/commits/d7d8f0b2712238610365019cb53d57062b480470)

**Comment by ariesdevil** [comment]
> > > Nice shot!
> > 
> > 
> > the problem is more complex than I think... After deletion, a `MetaServiceError` will be thrown from the drop-on-unknown DB test instead. The implementation of `create` and `drop` has a lot in common, but why `create` returns an `UnknowDataBase` while `drop` returns a `MetaServiceError`? This problem has only shown up in tests of `metasrv`, a further looking into is needed... I'll take a look tomorrow (this morning).
> 
> That should be incorrect error type that is used when a database is not found. Let me have a look.

@ClSlaid plz wait for #4182 merged then rebase it and I think all tests will be passed.

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**